src/components/billboard.js

```diff
@@ -0,0 +1,66 @@
+const isMobileVR = AFRAME.utils.device.isMobileVR();
+
+// Billboard component that only updates visible objects and only those in the camera view on mobile VR.
+AFRAME.registerComponent("billboard", {
+  init: function() {
+    this.target = new THREE.Vector3();
+    this._updateBillboard = this._updateBillboard.bind(this);
+    this._updateIsInView = this._updateIsInView.bind(this);
+
+    if (isMobileVR) {
+      this.el.sceneEl.systems["frame-scheduler"].schedule(this._updateIsInView, "billboards");
+    }
```

Comment:
Doing this only one per frame seems pretty bad to me. It means that if there are 20 people in the room, you could spend half a second maybe looking at someone before their nametag suddenly rotates to face you. I feel that will seem like a bug.

Reply:
I agree this is ultimately the tradeoff in exchange for a higher frame rate on mobile VR, we are taking on that visual impact. Having multiple pieces of work per frame or a better regulator would be an improvement but in general our approach to spreading across work is a one-per-frame within a scope model right now